integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -541,9 +543,11 @@ object CarbonDataRDDFactory {
           carbonLoadModel,
           loadStatus,
           newEntryLoadStatus,
-          overwriteTable)
+          overwriteTable,
+          uniqueTableStatusId)
+      PreAggregateUtil.commitDataMaps(carbonLoadModel, uniqueTableStatusId)(sqlContext.sparkSession)
```

Comment:
Add a listener for postStatusUpdateEvent and commit in that listner

Reply:
Use try catch and make status fail in case of any issue while committing the datamap